src/shogun/mathematics/linalg/LinalgBackendBase.h

```diff
@@ -28,69 +27,63 @@
  * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
  * POSSIBILITY OF SUCH DAMAGE.
  *
- * Authors: 2016 Pan Deng, Soumyajit De, Viktor Gal
+ * Authors: 2016 Pan Deng, Soumyajit De, Heiko Strathmann, Viktor Gal
  */
 
 #include <shogun/lib/config.h>
-#include <shogun/io/SGIO.h>
+#include <shogun/lib/common.h>
 #include <shogun/lib/SGVector.h>
-#include <shogun/mathematics/linalg/BaseVector.h>
+#include <shogun/io/SGIO.h>
+#include <shogun/mathematics/linalg/GPUMemoryBase.h>
 #include <memory>
 
-#ifndef GPU_Vector_H__
-#define GPU_Vector_H__
-
-#ifdef HAVE_CXX11
+#ifndef Linalg_Backend_Base_H__
+#define Linalg_Backend_Base_H__
 
 namespace shogun
 {
 
-/** GPU vector wrapper structure */
-template <class T>
-struct GPUVector : public BaseVector<T>
+/** Base class for generic linalg methods */
+class LinalgBackendBase
 {
-	/** Structure for different types of GPU vectors */
-	struct GPUArray;
-
-	/** Opaque pointer of GPUArray */
-	std::unique_ptr<GPUArray> gpuarray;
-
-	/** Default Constructor */
-	GPUVector();
-
-	/** Wrap SGVector */
-	GPUVector(const SGVector<T> &vector);
-
-	/** Copy Constructor*/
-	GPUVector(const GPUVector<T> &vector);
+public:
+	/** Generic vector dot-product macro */
```

Comment:
You can reference the base interface here and say this is the implementation wrapper of that. Note: This doc is never read by users, only by developers, so it should give the appropriate information